test/correctness/async.cpp

```diff
@@ -0,0 +1,447 @@
+#include "Halide.h"
+
+using namespace Halide;
+
+#ifdef _WIN32
+#define DLLEXPORT __declspec(dllexport)
+#else
+#define DLLEXPORT
+#endif
+
+extern "C" DLLEXPORT int expensive(int x) {
+    float f = 3.0f;
+    for (int i = 0; i < (1 << 10); i++) {
+        f = sqrtf(sinf(cosf(f)));
+    }
+    if (f < 0) return 3;
+    return x;
+}
+HalideExtern_1(int, expensive, int);
+
+int main(int argc, char **argv) {
+
+    // Basic compute-root async producer
+    if (1) {
```

Comment:
I presume the `if (1)` are for testing/debugging convenience, but we should probably remove them before landing